hail/python/hail/experimental/ndarray.py

```diff
@@ -0,0 +1,16 @@
+import hail as hl
+import numpy as np
+import hail.ir as ir
+from hail.expr import construct_expr, to_expr
+
+from hail.typecheck import typecheck
+
+
+@typecheck(nd=np.ndarray)
```

Comment:
do we not have expr_ndarray or the equivalent anymore?

Reply:
There is a stub for `expr_ndarray`. My initial thought was to do all expression integration in the next PR, but this does feel a little half-baked. Would you like me to just add that to this PR?

Reply:
Ok so integrating with expressions is going to require some decisions and potentially a decent amount of work:
- We would have to make a decision about how to coerce lists now. Should `[[1, 2], [3, 4]]` be an array of arrays or a ndarray? The current typecheck architecture blindly coerces inputs assuming there doesn't exist this type of discrepancy. With the current structure, to get an NDArray from nested python lists they would first be converted to arrays of arrays and we would then have to recursively flatten the array expressions.
- I changed `NDArrayExpression` to only extend `Expression` not `CollectionExpression` because `CollectionExpression` tries to treat its subclasses as `Array`s which is not supported by `NDArray`. Will have to come back to this as well when figuring out how to integrate `NDArray`s with `Array`s

Talked to Cotton and decided to keep it simple to start, with `_ndarray` constructing an `NDArrayExpression` from python lists and numpy ndarrays, _not_ other expressions.